pkg/util/export/etl/db/db_holder.go

```diff
@@ -254,162 +254,87 @@ func getPrepareSQL(tbl *table.Table, columns int, batchLen int, middleBatchLen i
-					}
-					idx++
-				}
-			}
-			_, err := maxStmt.ExecContext(ctx, vals...)
-			if err != nil {
-				tx.Rollback()
-				return err
-			}
-			if ctx.Err() != nil {
-				tx.Rollback()
-				return ctx.Err()
-			}
-
-			records = records[batchLen:]
-		} else if len(records) >= middleBatchLen {
-			if maxStmt != nil {
-				err = maxStmt.Close()
-				if err != nil {
-					tx.Rollback()
-					return err
-				}
-				maxStmt = nil
-			}
-			if middleStmt == nil {
-				middleStmt, err = tx.PrepareContext(ctx, sqls.middleRows)
-				if err != nil {
-					tx.Rollback()
-					return err
-				}
-			}
-			vals := make([]any, sqls.columns*middleBatchLen)
-			idx := 0
-			for _, row := range records[:middleBatchLen] {
-				for i, field := range row {
-					escapedStr := field
-					if tbl.Columns[i].ColType == table.TVarchar && tbl.Columns[i].Scale < len(escapedStr) {
-						vals[idx] = field[:tbl.Columns[i].Scale-1]
-					} else {
-						vals[idx] = field
-					}
-					idx++
-				}
-			}
-			_, err := middleStmt.ExecContext(ctx, vals...)
-			if err != nil {
-				tx.Rollback()
-				return err
-			}
-			if ctx.Err() != nil {
-				tx.Rollback()
-				return ctx.Err()
-			}
-
-			records = records[middleBatchLen:]
-		} else {
-			if maxStmt != nil {
-				err = maxStmt.Close()
-				if err != nil {
-					tx.Rollback()
-					return err
-				}
-				maxStmt = nil
-			}
-			if middleStmt != nil {
-				err = middleStmt.Close()
-				if err != nil {
-					tx.Rollback()
-					return err
-				}
-				middleStmt = nil
-			}
-			if oneStmt == nil {
-				oneStmt, err = tx.PrepareContext(ctx, sqls.oneRow)
-				if err != nil {
-					tx.Rollback()
-					return err
-				}
-			}
-			vals := make([]any, sqls.columns)
-			for _, row := range records {
-				if err != nil {
-					return err
-				}
-				for i, field := range row {
-					escapedStr := field
-					if tbl.Columns[i].ColType == table.TVarchar && tbl.Columns[i].Scale < len(escapedStr) {
-						vals[i] = field[:tbl.Columns[i].Scale-1]
-					} else {
-						vals[i] = field
-					}
-				}
-				_, err = oneStmt.ExecContext(ctx, vals...)
-				if err != nil {
-					tx.Rollback()
-					return err
-				}
-			}
-			err = oneStmt.Close()
-			if err != nil {
-				tx.Rollback()
-				return err
-			}
-			break
+func (w *CSVWriter) GetContent() string {
+	w.formatter.Flush() // Ensure all data is written to buffer
+	return w.buf.String()
+}
+
+func (w *CSVWriter) Release() {
+	if w.buf != nil {
+		w.buf.Reset()
+		w.buf = nil
+		w.formatter = nil
+	}
+	putBuffer(w.buf)
+}
+
+func bulkInsert(ctx context.Context, sqlDb *sql.DB, records [][]string, tbl *table.Table, batchLen int, middleBatchLen int) error {
+	if len(records) == 0 {
+		return nil
+	}
+
+	csvWriter := NewCSVWriter(ctx)
+	defer csvWriter.Release() // Ensures that the buffer is returned to the pool
+
+	// Write each record to the CSVWriter
+	for _, record := range records {
+		if err := csvWriter.WriteStrings(record); err != nil {
+			return err
 		}
 	}
 
```

Comment:
Here, can you add one more line to the csvData, basically that line should be equivalent as if  it is
log.Infof("Log wrote %d records %d bytes", x, y)

Reply:
resolved as discussed